pkg/controller/service/controller.go

```diff
@@ -143,6 +146,9 @@ func New(
 		nodeLister:       nodeInformer.Lister(),
 		nodeListerSynced: nodeInformer.Informer().HasSynced,
 		queue:            workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "service"),
+		// feature gates checked in Run(), otherwise unit tests panic since feature gates are not known
+		legacyNodeRoleFeatureEnabled:       true,
```

Comment:
Why not pass these values as arguments to `New`?
Defaulting here decouples the value from any defaults defined in feature gates, which might result in an out-of-sync definition later on.

Reply:
I agree, actually just updated the PR so we don't need to default here anymore.